### PR TITLE
Replace google-cloud-ops-agent.target with google-cloud-ops-agent.service

### PR DESCRIPTION
**Pull request by quentinmit**
Fixes b/185809793 and b/191712335

### CODE REVIEW

**Comment by quentinmit** [comment]
I've manually tested the following flows:
- fresh install
- upgrade from 1.x
- upgrade from 1.x with disabled target

on:
- Debian 9
- Debian 10
- SLES 12

and got the expected behavior each time (old daemons not running, new daemons running if new or enabled before)

**Review by qingling128 — APPROVED** [review]
LGTM

File: debian/preinst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+set -e
+
+# Upgrade from <2.0 needs to stop the .target
```
Comment (igorpeshansky) [inline debian/preinst:4]: Nit: comments in postinst have trailing `.`s…

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
One minor comment nit, otherwise LGTM.

**Comment by igorpeshansky** [comment]
Actually, I just tested the upgrade path on CentOS 7, and it doesn't seem to work:
```
$ sudo yum install google-cloud-ops-agent
…
Running transaction
  Updating   : google-cloud-ops-agent-2.0.0-1.el7.x86_64                                                                                                            1/2 
  Cleanup    : google-cloud-ops-agent-1.0.10-1.el7.x86_64                                                                                                           2/2 
  Verifying  : google-cloud-ops-agent-2.0.0-1.el7.x86_64                                                                                                            1/2 
  Verifying  : google-cloud-ops-agent-1.0.10-1.el7.x86_64                                                                                                           2/2 

Updated:
  google-cloud-ops-agent.x86_64 0:2.0.0-1.el7                                                                                                                           

Complete!
$ systemctl status google-cloud-ops-agent\*
● google-cloud-ops-agent-collectd.service
   Loaded: not-found (Reason: No such file or directory)
   Active: active (running) since Tue 2021-06-29 19:50:57 UTC; 21min ago
…
```

**Comment by igorpeshansky** [comment]
f59acabc59d9eb477ca870ca6ea630bb6a6cf231 worked on CentOS 7.

File: debian/preinst (hunk 1 above)
Comment (quentinmit) [reply debian/preinst:4]: This was a conscious choice to avoid abiguity on the filename.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

File: systemd/google-cloud-ops-agent.target [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-[Unit]
-Description=Google Cloud Ops Agent
-Requires=google-cloud-ops-agent.service google-cloud-ops-agent-fluent-bit.service google-cloud-ops-agent-opentelemetry-collector.service
-After=google-cloud-ops-agent.service google-cloud-ops-agent-fluent-bit.service google-cloud-ops-agent-opentelemetry-collector.service
```
Comment (sophieyfang) [inline systemd/google-cloud-ops-agent.target:18]: Add this line back in `ops-agent.service`- Follow up PR: https://github.com/GoogleCloudPlatform/ops-agent/pull/129

**Comment by shibumi** [comment]
Hey,

Can someone provide more information why the service target has been transformed into a systemd oneshot service with dependencies? This is pretty much against all systemd best practices and I would really like to understand the reasoning behind that decision.

**Pull request merged**